clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1208,25 +1246,43 @@ public void maybeRegisterConnectionMetrics(String connectionId) {
             }
         }
 
-        public void recordBytesSent(String connectionId, long bytes) {
-            long now = time.milliseconds();
-            this.bytesSent.record(bytes, now);
+        public void recordPartialSend(String connectionId, long bytes, long currentTimeMs) {
```

Comment:
`recordPartialSend` and `recordCompletedSend` sound like they are disjoint sets. But actually `PartialSend` could also be a `CompletedSend`, we are just recording different metrics. Perhaps we could rename `recordPartialSend` to reflect that it is recording bytes sent?

Reply:
Makes sense. Thanks for the suggestion.